### PR TITLE
Added support for SFTP with public/private key

### PR DESCRIPTION
**Pull request by cappelaere**
**Summary:** Summary of changes

Added Support for SFTP with public/private key

## Changes

* Detailed list or prose of changes
* ...

## Test Plan
Things that should succeed before merging.

- [ ] Unit tests
- [ x] Adhoc testing
- [ ] Update CHANGELOG
- [ ] Run `./bin/eslint-ratchet` and verify that eslint errors have not increased
  - [ ] Commit `.eslint-ratchet-high-water-mark` if the score has improved

## Release PR

If this is a **release** PR, make sure you branch name starts with `release-` prefix, e.g. `release-v-1.1.2`.

Reviewers: @tag-your-friends

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Please note: [Upgrade Instructions](https://cumulus-nasa.github.io/upgrade/1.7.0.html)
 
 ### Added
```
[the remark this answers is not in the document]
Comment (scisco) [reply CHANGELOG.md]: I like the idea of enabling the private key support on the test image (the one used in our circleci build). After enabling that you can stub out the KMS part for unit testing. For the integration tests however, we can probably use KMS.

[the remark this answers is not in the document]
Comment (laurenfrederick) [reply CHANGELOG.md]: I would say at the very least this needs to be unit tested. Changing the circle ci image is fine.

[the remark this answers is not in the document]
Comment (cappelaere) [reply CHANGELOG.md]: Tests have been provided by adding a new sftp service that supports public/private keys.  For testing, the keys are not encrypted.

**Comment by cappelaere** [comment]
Close #420

**Pull request merged**